tests/functional/mesh/mesh.go

```diff
@@ -0,0 +1,291 @@
+			// Update this to choose which listener to dial into
+			if MeshDefinition.Nodes[conn].Listen[0].Protocol == "tcp" {
+				node.TCPDial(MeshDefinition.Nodes[conn].Listen[0].Addr, cost)
+			} else if MeshDefinition.Nodes[conn].Listen[0].Protocol == "udp" {
+				node.UDPDial(MeshDefinition.Nodes[conn].Listen[0].Addr, cost)
+			}
+		}
+	}
+	return &Mesh{
+		Nodes,
+		MeshDefinition,
+	}, nil
+}
+
+// Shutdown This is broken and causes the thread to hang, dont use until
+// netceptor.Shutdown is fixed
+func (m *Mesh) Shutdown() {
+	for _, node := range m.Nodes {
+		node.NetceptorInstance.Shutdown()
+	}
+}
+
+// CheckConnections returns true if the connections defined in our mesh definition are
+// consistent with the connections made by the nodes
+func (m *Mesh) CheckConnections() bool {
+	for _, status := range m.Status() {
+		actualConnections := map[string]float64{}
+		for _, connection := range status.Connections {
+			actualConnections[connection.NodeID] = connection.Cost
+		}
+		expectedConnections := map[string]float64{}
+		for k, v := range m.MeshDefinition.Nodes[status.NodeID].Connections {
+			expectedConnections[k] = v
+		}
+		for nodeID, node := range m.MeshDefinition.Nodes {
+			if nodeID == status.NodeID {
+				continue
+			}
+			for k, v := range node.Connections {
+				if k == status.NodeID {
+					expectedConnections[nodeID] = v
+				}
+			}
+		}
+		if reflect.DeepEqual(actualConnections, expectedConnections) {
+			return true
+		}
+	}
+	return false
+}
+
+// CheckKnownConnectionCosts returns true if every node has the same view of the connections in the mesh,
+// if they do, we can assume the routing is consistent across all nodes
+func (m *Mesh) CheckKnownConnectionCosts() bool {
+	meshStatus := m.Status()
+	// If the mesh is empty we are done
+	if len(meshStatus) == 0 {
+		return true
+	}
+
+	knownConnectionCosts := meshStatus[0].KnownConnectionCosts
+	for _, status := range m.Status() {
+		if !reflect.DeepEqual(status.KnownConnectionCosts, knownConnectionCosts) {
+			return false
+		}
+	}
+	return true
+}
+
+// WaitForReady Waits for connections and routes to converge
+func (m *Mesh) WaitForReady(timeout float64) error {
+	// TODO: Add a poll interval parameter
+	connectionsReady := m.CheckConnections()
+	for ; timeout > 0 && !connectionsReady; connectionsReady = m.CheckConnections() {
+		time.Sleep(100 * time.Millisecond)
+		timeout -= 100
+	}
+	if connectionsReady == false {
+		return errors.New("Timed out while waiting for connections")
+	}
+
+	routesReady := m.CheckKnownConnectionCosts()
+	for ; timeout > 0 && !routesReady; routesReady = m.CheckKnownConnectionCosts() {
+		time.Sleep(100 * time.Millisecond)
+		timeout -= 100
+	}
+	if routesReady == false {
+		return errors.New("Timed out while waiting for routes to settle")
+	}
+
+	return nil
+}
+
+// Status returns a list of statuses from the contained netceptors
+func (m *Mesh) Status() []netceptor.Status {
+	out := []netceptor.Status{}
```

Comment:
```suggestion
	var out []netceptor.Status
```
avoids an unnecessary allocation